packages/rum-recorder/src/domain/rrweb/mutation.ts

```diff
@@ -300,19 +296,24 @@ export class MutationObserverWrapper {
       attributes: this.attributes
         .map((attribute) => ({
           attributes: attribute.attributes,
-          id: mirror.getId(attribute.node as INode),
+          id: getSerializedNodeId(attribute.node),
         }))
-        // attribute mutation's id was not in the mirror map means the target node has been removed
-        .filter((attribute) => mirror.has(attribute.id)),
+        // ignore mutations whose target node has been removed
+        .filter((attribute) => attribute.id !== -1),
       removes: this.removes,
       texts: this.texts
         .map((text) => ({
-          id: mirror.getId(text.node as INode),
+          id: getSerializedNodeId(text.node),
           value: text.value,
         }))
-        // text mutation's id was not in the mirror map means the target node has been removed
-        .filter((text) => mirror.has(text.id)),
+        // ignore mutations whose target node has been removed
+        .filter((text) => text.id !== -1),
     }
+
+    while (this.mapRemoves.length) {
+      recursivelyRemoveSerializedNodes(this.mapRemoves.shift()!)
+    }
```

Comment:
For my understanding, why do we need to remove serialized nodes from removed nodes?